pkg/scheduler/factory/factory.go

```diff
@@ -262,10 +267,17 @@ func NewConfigFactory(args *ConfigFactoryArgs) Configurator {
 	}
 
 	// storageClassInformer is only enabled through VolumeScheduling feature gate
-	var storageClassLister storagelisters.StorageClassLister
+	var storageClassLister storagelistersv1.StorageClassLister
 	if args.StorageClassInformer != nil {
 		storageClassLister = args.StorageClassInformer.Lister()
 	}
+
+	// csiNodeInformer is only enabled through AttachVolumeLimit feature gate
+	var csiNodeLister storagelistersv1beta1.CSINodeLister
+	if args.CSINodeInformer != nil {
```

Comment:
Where are we passing nil for CSINodeInformer, if featuregates are disabled?

Reply:
+1

Reply:
I think the comment above is not accurate. Removed.

I'm keeping the check because the informer is an interface and theory could be `nil` (the `StorageClassInformer` above also checks that).